core/src/main/java/org/frankframework/http/HttpSender.java

```diff
@@ -270,37 +244,47 @@ protected HttpRequestBase getMethod(URI uri, Message message, ParameterValueList
 	 */
 	private HttpPost getMultipartPostMethodWithParamsInBody(URI uri, Message message, ParameterValueList parameters, PipeLineSession session) throws SenderException, IOException {
 		HttpPost hmethod = new HttpPost(uri);
+		hmethod.setEntity(createHttpEntity(message, parameters, session));
+		return hmethod;
+	}
 
+	private HttpEntity createHttpEntity(Message message, ParameterValueList parameters, PipeLineSession session) throws IOException, SenderException {
+		HttpEntity httpEntity;
 		if (postType==PostType.URLENCODED && StringUtils.isEmpty(getMultipartXmlSessionKey())) { // x-www-form-urlencoded
-			List<NameValuePair> requestFormElements = new ArrayList<>();
+			httpEntity = createUrlEncodedFormEntity(message, parameters);
+		}
+		else { //formdata and mtom
+			httpEntity = createMultiPartEntity(message, parameters, session);
+		}
+		return httpEntity;
```

Comment:
Waarom return je niet direct de entity?

Reply:
Ik dacht er aan dat te doen en deed het toen toch weer niet. Geen echt goede reden ervoor.
Ik kan het in een volgend PR doen want ik heb nog een paar andere dingen die ik wil versimpelen in de code, maar die wou ik niet meenemen in dit PR om het overzichtelijker te houden.

Reply:
'4. Early returns' ;-) Mee eens